### ISSUE TITLE
SPARQL Graph - user interface glitch / feature interaction

### ISSUE THREAD

**Issue by Robert-Adelard** [ge-high-assurance/RACK#500]
If you copy part of the SPARQL query generated from a graph, SPARQL Graph considers the SPARQL to have been modified and prompts you to download it before making any further modifications to the graph.

This is mildly annoying but generally just a nuisance if you are clicking on a property - however, if you drag a new node onto the canvas, there is a surprising feature interaction...

1. The prompt to download the SPARQL appears and you click Clear
2. Nothing happens, so you click Clear again (and perhaps again)
3. After a while, the path search times out, and multiple select path dialogs appear

The problem is that the warning about the modified SPARQL does not disappear when you select Clear if a path search is in progress, so you are tempted to click Clear again. However, SPARQL Graph seems to interpret subsequent mouse clicks as node drag and drop requests, which is why multiple path dialogs appear.

To summarise, I think there are two bugs here:

1. If possible, SPARQL Graph should distinguish between copying and editing the query, so that it doesn't generate spurious warnings about the query having been modified
2. If the warning appears as a result of a node drag and drop,  it should be dismissed before the path search

**Comment by cuddihyge** [comment]
Have tried many different ways to reproduce this and failed.
Since it is a year old, I'm presuming that intervening improvements have fixed this.
Instead of closing, I'll move it to Ready-to-review, and Robert can retry with the next dev version and comment if needed.
